src/routes/+page.server.js

```diff
@@ -1,12 +1,107 @@
-import { get_concepts } from '$lib/server/ontology'
+import { get_concepts, get_filtered_simplification_hints } from '$lib/server/ontology'
 
 /** @type {import('./$types').PageServerLoad} */
 export async function load({ url: { searchParams }, locals: { db_ontology } }) {
-	const search_criteria = Object.fromEntries(searchParams)
+	/** @type {ConceptSearchFilter} */
+	const search_filter = {
+		q: '',
+		scope: 'stems',
+		category: '',
+		...Object.fromEntries(searchParams),
+	}
+
+	const matches = await get_concepts(db_ontology)(search_filter)
+	const how_to_matches = await get_filtered_simplification_hints(db_ontology)(search_filter)
+
+	// filter out the how-to matches that are not associated with a concept
+	const unlinked_how_to_matches = how_to_matches.filter(hint =>
+		!matches.some(match => `${match.stem}-${match.sense}` === hint.term && hint.part_of_speech === match.part_of_speech))
 
-	const matches = await get_concepts(db_ontology)(search_criteria)
+	// if a remaining how-to result has a sense, it is pending entry into the ontology and should be treated as a concept
+	const pending_results = unlinked_how_to_matches.filter(hint => ends_in_a_sense(hint.term)).map(create_pending_result)
+	const how_to_hints = unlinked_how_to_matches.filter(hint => !ends_in_a_sense(hint.term))
+	
+	// group the remaining entries by term and part_of_speech, so we can show all hints for a given term together
+	const how_to_results = [...new Set(how_to_hints.map(hint => `${hint.term}|${hint.part_of_speech}`))].map(create_how_to_result)
+
+	const results = [
+		...matches,
+		...pending_results,
+		...how_to_results,
+	]
 
 	return {
-		matches,
+		results,
+	}
+
+	/**
+	 * @param {string} term
+	 * @returns {boolean}
+	 */
+	function ends_in_a_sense(term) {
+		const ENDS_IN_A_SENSE = /-[A-Z]$/
```

Comment:
I saw this same regex over in another file, is it worthwhile (yet) to create another file for these?